packages/consumption/src/modules/requests/itemProcessors/readAttribute/ReadAttributeRequestItemProcessor.ts

```diff
@@ -53,21 +55,100 @@ export class ReadAttributeRequestItemProcessor extends GenericRequestItemProcess
+                        requestInfo
+                    );
+                    return AttributeSuccessionAcceptResponseItem.from({
+                        result: ResponseItemResult.Accepted,
+                        successorId: successorOwnSharedAttribute.id,
+                        successorContent: successorOwnSharedAttribute.content,
+                        predecessorId: predecessorOwnSharedAttribute.id
+                    });
+                }
+                if (existingSourceAttribute.isRelationshipAttribute()) {
+                    if (existingSourceAttribute.isOwnedBy(this.accountController.identity.address)) {
+                        const successorOwnSharedAttribute = await this.performOwnSharedThirdPartyRelationshipAttributeSuccession(
+                            predecessorOwnSharedAttribute.id,
+                            existingSourceAttribute,
+                            requestInfo
+                        );
+                        return AttributeSuccessionAcceptResponseItem.from({
+                            result: ResponseItemResult.Accepted,
+                            successorId: successorOwnSharedAttribute.id,
+                            successorContent: successorOwnSharedAttribute.content,
+                            predecessorId: predecessorOwnSharedAttribute.id
+                        });
+                    }
+
+                    const successorOwnSharedAttribute = await this.performThirdPartyOwnedRelationshipAttributeSuccession(
+                        predecessorOwnSharedAttribute.id,
+                        existingSourceAttribute,
+                        requestInfo
+                    );
+                    return AttributeSuccessionAcceptResponseItem.from({
+                        result: ResponseItemResult.Accepted,
+                        successorId: successorOwnSharedAttribute.id,
+                        successorContent: successorOwnSharedAttribute.content,
+                        predecessorId: predecessorOwnSharedAttribute.id
+                    });
+                }
+            } else {
+                throw new Error("You cannot share the predecessor of an already shared Attribute version.");
+            }
         } else {
             sharedLocalAttribute = await this.createNewAttribute(parsedParams.newAttribute!, requestInfo);
+            return ReadAttributeAcceptResponseItem.from({
+                result: ResponseItemResult.Accepted,
+                attributeId: sharedLocalAttribute.id,
+                attribute: sharedLocalAttribute.content
+            });
         }
 
-        return ReadAttributeAcceptResponseItem.from({
-            result: ResponseItemResult.Accepted,
-            attributeId: sharedLocalAttribute.id,
-            attribute: sharedLocalAttribute.content
-        });
+        // this should never be reached
+        throw new Error("The ReadAttributeRequestItem must be either answered with an existing or a new Attribute.");
```

Comment:
you can just remove the else block instead of throwing this error (serval makes sure that this is valid)